File: langgraph/result/qwen/3b-py2js/116.js
function sortArray(arr) {
    return arr.sort((a, b) => {
        const countOnesA = bin(a).slice(2).split('').filter(char => char === '1').length;
        const countOnesB = bin(b).slice(2).split('').filter(char => char === '1').length;
        return countOnesA - countOnesB;
    });
}

function bin(n) {
    return n.toString(2);
}

File: langgraph/data/semantic_equiv/116.py
def arrange_numbers_by_binary_ones(numbers):
    def count_ones_in_binary(num):
        return bin(num).count('1')

    sorted_numbers = sorted(numbers)
    final_sorted_numbers = sorted(sorted_numbers, key=count_ones_in_binary)

    return final_sorted_numbers
# method name: sort_array -> arrange_numbers_by_binary_ones
# parameter name: arr -> numbers
# introduced a helper function count_ones_in_binary to count '1's in binary representation
# split the sorting process into two steps for clarity
# used the helper function in the second sort call